src/particle.cpp

```diff
@@ -353,6 +358,40 @@ void Particle::event_death()
   }
 }
 
+void Particle::energy_delivert_in_cell()
+{
+  // Update pht_storage array
+  pht_storage()[coord(0).cell] += E_last() - E();
+}
+
+void Particle::killed_particle_energy_delivert()
+{
+  // Add the energy of the killed particles to the pht_storage array
+  pht_storage()[coord(0).cell] += E();
```

Comment:
Similarly, we should probably not have this as a separate function and instead put it in `Particle::event_death`. Moreover, this will also not work with tallies based on `cell_instance`.

Reply:
I think that's part of the problem: my implementation doesn't yet work properly with OpenMC's tally functions. I think I need to work on how the tally output is determined in the end from the pht_storage array. Then we don't need to define 
external std::vector<double> bins_pht; 
nor
external int cell_pht;
here.
I think I was in too much of a hurry here overall and really wanted to create a reasonably running version.

Reply:
- `killed_particle_energy_delivered()` is now called at the end of `Particle::event_death`. But we could just remove the entire function and simply call the single line 
```cpp
pht_storage()[coord(0).cell] += E();
```
- Again, I need to think a bit on how to make this work with the `cell_instance` feature.